src/lib/optimizer/strategy/predicate_placement_rule.cpp

```diff
@@ -112,9 +113,7 @@ void PredicatePlacementRule::_push_down_traversal(const std::shared_ptr<Abstract
       auto right_push_down_nodes = std::vector<std::shared_ptr<AbstractLQPNode>>{};
 
       // It is safe to move predicates down past the named joins as doing so does not affect the presence of NULLs
-      if (join_node->join_mode == JoinMode::Inner || join_node->join_mode == JoinMode::Cross ||
-          join_node->join_mode == JoinMode::Semi || join_node->join_mode == JoinMode::AntiNullAsTrue ||
```

Comment:
Does this mean we do not push predicates below semi joins (etc.) anymore?

Reply:
No, see lines 108-109:

```cpp
      // Not a semi / anti join. We need to check if we can push the nodes in push_down_nodes past the join or if they
      // need to be inserted here before proceeding.
```

Semi and anti joins are already handled at this point.

Reply:
Ah okay :)